src/upload-button.tsx

```diff
@@ -0,0 +1,356 @@
+                    cancelledUploads.push(file);
+                }
+            } finally {
+                setUploadedFiles(oldFiles => {
+                    const copy = { ...oldFiles };
+                    delete copy[file.name];
+                    return copy;
+                });
+            }
+        }));
+
+        resetInput();
+        window.removeEventListener("beforeunload", beforeUnloadHandler);
+
+        // If all uploads are cancelled, don't show an alert
+        if (cancelledUploads.length !== toUploadFiles.length) {
+            addAlert(_("Upload complete"), AlertVariant.success, "upload-success", _("Succesfully uploaded file(s)"));
+        }
+    };
+
+    const isUploading = Object.keys(uploadedFiles).length !== 0;
+    let icon;
+
+    if (isUploading) {
+        let totalSize = 0;
+        let totalSent = 0;
+
+        Object.keys(uploadedFiles).forEach((key) => {
+            const uploadedFile = uploadedFiles[key];
+            totalSize += uploadedFile.file.size;
+            totalSent += uploadedFile.progress;
+        });
+
+        const overallProgress = ((totalSent / totalSize) * 100).toFixed(2);
+
+        icon = (
+            <Button
+              onClick={() => setPopover(true)}
+              className="progress-wrapper"
+              variant="plain"
+              icon={
+                  <div
+                    id="upload-progress-btn"
+                    className="progress-pie"
+                    title={`Upload ${overallProgress}% completed`}
+                    style={{ "--progress": `${overallProgress}%` } as React.CSSProperties}
+                  />
+              }
+            />
+        );
+    }
+
+    return (
+        <>
+            <Popover
+              className="upload-popover"
+              position={PopoverPosition.bottom}
+              headerContent={<p>{_("Uploads")}</p>}
+              bodyContent={Object.keys(uploadedFiles).map((key, index) => {
+                  const file = uploadedFiles[key];
+                  return (
+                      <React.Fragment key={index}>
+                          <Divider />
+                          <Flex className="upload-progress-flex" flexWrap={{ default: 'nowrap' }}>
+                              <Progress
+                                key={file.file.name}
+                                className={`upload-progress-${index} upload-progress pf-v5-m-tabular-nums`}
+                                value={file.progress}
+                                title={file.file.name}
+                                max={file.file.size}
+                              />
+                              <Button
+                                variant="plain"
+                                icon={<TrashIcon />}
+                                className={`cancel-button-${index} cancel-button`}
+                                onClick={file.cancel}
+                              />
```

Comment:
Shouldn't that get some kind of alt-text/aria-label?

Reply:
I think we have tons of these buttons without aria-label or alt-text? So I don't know.

Reply:
And our designer keeps telling us to not use `aria-label`'s without a reason as they are most of the time wrong. 

So following the PF guidelines:
```
Provide unique, descriptive text content to the button, or an aria-label if the button does not contain any visible text
```

I'll add a label.

Reply:
Well, he said "if there's no better way", right? Adding an actual text would make the popup rather cramped, so this may apply here.